### PR TITLE
Update QuestionnaireResponse Upon Input Change

### PR DESCRIPTION
**Pull request by mickohanlon23**
Created a button that allows the user to generate a QeustionnaireResponse after form modifications have been made. Clicking the "Generate Questionnaire Response" button should:
1) Read the rendered form in the HTML element and generate a corresponding QuestionnaireResponse
2) Log the questionnaire to the console
3) Print a fun little message below the button

**To Test:**
I modified the `sample-patient-bundle` so that two CancerRelatedSurgicalProcedures exist. Clicking the "Generate Questionnaire Response" button without selecting any answer options should console.log() a questionnaire response with only one item containing an answer. If you then choose one of the answer options for CancerRelatedSurgicalProcedure, and re-click the button, the console should log a new questionnaire response containing two items with answers, since you have chosen a CancerRelatedSurgicalProcedure answer.

*Note: I left the `sample-patient-bundle` modifications here so that it can be tested, but I can delete the extra resource after this is reviewed.*

### CODE REVIEW

**Comment by pmadria** [comment]
Looks good! Functionality works as intended.

**Review by pmadria — COMMENTED** [review]
Looks good and Functionality works as intended!

**Review by pmadria — APPROVED** [review]

File: src/components/Abstractor/Abstractor.tsx [hunk 1]
```diff
@@ -30,16 +30,35 @@ const Abstractor = ({ patientData, library, valueSetMap, questionnaire }: Props)
...
+    let answerCount: number = 0;
+    if (qr.item){
+      answerCount = qr.item.length;
+    }
+    
+    const response: HTMLElement | null = document.getElementById("responseGenerated");
+    if (response){
+      response!.innerHTML = `Questionnaire Response has been generated with ${answerCount} answer(s) and has been logged to the console!`;
+    }
+  }
```
Comment (mgramigna) [inline src/components/Abstractor/Abstractor.tsx:53]: There are very few situations where we should set the innerHTML of something in React. Generally, if the state of something updates (in this case, the face that a response has been generated.

What you should do instead is this:

1) Create a new state variable using the useState hook:

``` Typescript
const [responseGenerated, setResponseGenerated] = useState<boolean>(false);
```

2) When the button is clicked, update the state:

``` Typescript
setResponseGenerated(true);
```

3) In the jsx, display the message if the variable is true:

``` jsx
{responseGenerated && <p>Questionnaire Response has been generated and has been logged to the console!</p>
```

**Review by mgramigna — CHANGES_REQUESTED** [review]

**Comment by mickohanlon23** [comment]
> Created a button that allows the user to generate a QeustionnaireResponse after form modifications have been made. Clicking the "Generate Questionnaire Response" button should:
> 
> 1. Read the rendered form in the HTML element and generate a corresponding QuestionnaireResponse
> 2. Log the questionnaire to the console
> 3. Print a fun little message below the button
> 
> **To Test:**
> I modified the `sample-patient-bundle` so that two CancerRelatedSurgicalProcedures exist. Clicking the "Generate Questionnaire Response" button without selecting any answer options should console.log() a questionnaire response with only one item containing an answer. If you then choose one of the answer options for CancerRelatedSurgicalProcedure, and re-click the button, the console should log a new questionnaire response containing two items with answers, since you have chosen a CancerRelatedSurgicalProcedure answer.
> 
> _Note: I left the `sample-patient-bundle` modifications here so that it can be tested, but I can delete the extra resource after this is reviewed._

Refactored HTML elements by using useState hook. Functionality should remain the same. Restored the original patient bundle.
Clicking the "Generate Questionnaire" button without editing the form should return a Questionnaire Response with 2 answers. Can be tested by deleting one of the answer values in the form. Re-clicking the "Generate Questionnaire" button should return a Questionnaire Response with only 1 answer.

File: src/components/Abstractor/Abstractor.tsx [hunk 2]
```diff
@@ -30,16 +41,30 @@ const Abstractor = ({ patientData, library, valueSetMap, questionnaire }: Props)
...
+  const generateQR = () => {
+    // Generate QuestionnaireResponse
+    const qr = window.LForms.Util.getFormFHIRData('QuestionnaireResponse', 'R4');
+    console.log(qr);
+    // Signify to user that QuestionnaireResponse has been generated
+    let answerCount = 0;
+    if (qr.item) {
+      answerCount = qr.item.length;
+    }
+    setResponseGenerated(responseGenerated => ({...responseGenerated,  message: true, count: answerCount}) );
```
Comment (mgramigna) [inline src/components/Abstractor/Abstractor.tsx:58]: not sure how this doesn't give you an error, but the setResponseGenerated function sets the state variable for responseGenerated. Meaning the argument to it should be an object that matches the `ResponseObject` interface that you defined.

```suggestion
    setResponseGenerated({ count: answerCount, isGenerated: true });
```

**Review by mgramigna — CHANGES_REQUESTED** [review]
Nothing appears in the app per the following error

**Comment by mickohanlon23** [comment]
Interesting, I wasn't getting an error, and was still seeing the form. I did realize I forgot to update one of the property names so that the message wasn't generated after the button click. I accepted your changes and updated accordingly.

**Review by mgramigna — APPROVED** [review]

**Pull request merged**